source/agora/node/FullNode.d

```diff
@@ -153,30 +153,13 @@ public class FullNode : API
     ***************************************************************************/
 
     public void start ()
-    {
-        this.startPeriodicDiscovery();
-        this.network.startPeriodicCatchup(this.ledger);
-    }
-
-    /***************************************************************************
-
-        Starts the periodic network discovery task.
-
-    ***************************************************************************/
-
-    private void startPeriodicDiscovery ()
     {
         import core.time;
 
-        this.taskman.runTask(
-        ()
-        {
-            while (1)
-            {
-                this.network.discover();
-                this.taskman.wait(5.seconds);
-            }
-        });
+        void discover () { this.network.discover(); }
```

Comment:
> You don't need the startPeriodicDiscovery function anymore. And also:

Sorry I thought that we don't need the `runTask`, but actually it would make sense to run it inside a task to make it non blocking.

How about we use the same technique which you did in the Validator?

So `startPeriodicDiscovery` with a `runTask` inside, and start would call it first and then `startPeriodicCatchup`. Inside `startPeriodicDiscovery` you would have this `void discover` and the setting up of the timer.

Do you see what I mean?

Reply:
It's good to make the FullNode and Validator consistent.

Reply:
It's good to be consistent. It's been applied.